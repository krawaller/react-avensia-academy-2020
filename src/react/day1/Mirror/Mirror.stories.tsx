import React from 'react'
import { storiesOf } from '@storybook/react'
import { Mirror } from './Mirror'

storiesOf('day 1/Props/Mirror', module)
  .add('truthful', () => <Mirror truthful />) // same as <Mirror truthful={true} />
  .add('lying', () => <Mirror />) // same as <Mirror truthful={false} />
