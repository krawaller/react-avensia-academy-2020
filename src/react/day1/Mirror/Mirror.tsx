import React from 'react'

export type MirrorProps = {
  truthful?: boolean
}

export const Mirror = (props: MirrorProps) => {
  if (props.truthful) {
    return <p>The fairest in the land is Snow white!</p>
  } else {
    return <p>The fairest in the land is you my queen!</p>
  }
}

// ---- OTHER IMPLEMENTATION POSSIBILITIES -------

const Mirror2 = (props: MirrorProps) => {
  const fairest = props.truthful ? 'Snow white' : 'you my queen'
  return <p>The fairest in the land is {fairest}!</p>
}

const Mirror3 = ({ truthful }: MirrorProps) => (
  <p>The fairest in the land is {truthful ? 'Snow white' : 'you my queen'}!</p>
)
