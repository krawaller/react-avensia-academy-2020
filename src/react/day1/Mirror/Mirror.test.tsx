import React from 'react'
import { render } from '@testing-library/react'
import { Mirror } from './Mirror'

describe('the Mirror component', () => {
  describe('using logical testing', () => {
    it('mentions snow white when truthful', () => {
      const { queryAllByText } = render(<Mirror truthful />)
      expect(queryAllByText(/snow ?white/i)).toBeTruthy()
      expect(queryAllByText('you')).toBeFalsy()
    })
    it('mentions "you" when lying', () => {
      const { queryAllByText } = render(<Mirror />)
      expect(queryAllByText('you')).toBeTruthy()
      expect(queryAllByText(/snow ?white/i)).toBeFalsy()
    })
  })
  describe('using snapshot testing', () => {
    it('renders as expected when lying', () => {
      const { container } = render(<Mirror />)
      expect(container).toMatchSnapshot()
    })
    it('renders as expected when truthful', () => {
      const { container } = render(<Mirror truthful />)
      expect(container).toMatchSnapshot()
    })
  })
})
