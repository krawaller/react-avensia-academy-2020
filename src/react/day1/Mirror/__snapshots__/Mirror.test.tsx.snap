// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Mirror component using snapshot testing renders as expected when lying 1`] = `
<div>
  <p>
    The fairest in the land is Snow white!
  </p>
</div>
`;

exports[`the Mirror component using snapshot testing renders as expected when truthful 1`] = `
<div>
  <p>
    The fairest in the land is you my queen!
  </p>
</div>
`;
